### PR TITLE
[Integration][PagerDuty] PORT-10680 Added default action to PagerDuty

### PR DESCRIPTION
**Pull request by oiadebayo**
# Description

**What -**  
Implemented a default "Trigger incident" action for the PagerDuty Integration. This action allows users to create a PagerDuty incident with user inputs for title and description, notifying the on-call team of critical issues.

**Why -**  
To improve the ease of creating and managing incidents within the PagerDuty integration. This default action helps users quickly trigger incidents with minimal configuration, driving higher engagement and adoption. It also leverages the new secret referencing feature for secure interactions via webhooks.

**How -**  
- Added a default action definition for triggering PagerDuty incidents with required fields such as title and description.
- Configured webhook invocation to trigger the incident creation in PagerDuty, with appropriate headers and secret references for secure API access.

## Type of change

Please leave one option from the following and delete the rest:

- [x] New feature (non-breaking change which adds functionality)

<h4> All tests should be run against the port production environment(using a testing org). </h4>

### Core testing checklist

- [ ] Integration able to create all default resources from scratch
- [ ] Resync finishes successfully
- [ ] Resync able to create entities
- [ ] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Scheduled resync able to abort existing resync and start a new one
- [ ] Tested with at least 2 integrations from scratch
- [ ] Tested with Kafka and Polling event listeners
- [ ] Tested deletion of entities that don't pass the selector

### Integration testing checklist

- [ ] Integration able to create all default resources from scratch
- [ ] Resync able to create entities
- [ ] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Resync finishes successfully
- [ ] If new resource kind is added or updated in the integration, add example raw data, mapping and expected result to the `examples` folder in the integration directory.
- [ ] If resource kind is updated, run the integration with the example data and check if the expected result is achieved
- [ ] If new resource kind is added or updated, validate that live-events for that resource are working as expected
- [ ] Docs PR link [here](#)

### Preflight checklist

- [ ] Handled rate limiting
- [ ] Handled pagination
- [ ] Implemented the code in async
- [ ] Support Multi account

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

## API Documentation

Provide links to the API documentation used for this integration.

### CODE REVIEW

File: integrations/pagerduty/.port/resources/actions.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    },
+    "invocationMethod": {
+      "type": "WEBHOOK",
+      "url": "https://api.pagerduty.com/incidents",
+      "agent": false,
+      "synchronized": true,
+      "method": "POST",
+      "headers": {
+        "RUN_ID": "{{ .run.id }}",
+        "Authorization": "Token token={{ .secrets.PAGERDUTY_API_TOKEN }}",
```
Comment (PeyGis) [inline integrations/pagerduty/.port/resources/actions.json:38]: let's have a unified secret naming using secrets.variable_name. I see we have different format for the snyk and jira

**Review by PeyGis — CHANGES_REQUESTED** [review]
looks good overall. left some minor comment. Please apply the same to the Jira action

File: integrations/pagerduty/.port/resources/actions.json (hunk 1 above)
Comment (oiadebayo) [reply integrations/pagerduty/.port/resources/actions.json]: Done

File: integrations/pagerduty/.port/resources/actions.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      "url": "https://api.pagerduty.com/incidents",
+      "agent": false,
+      "synchronized": true,
+      "method": "POST",
+      "headers": {
+        "RUN_ID": "{{ .run.id }}",
+        "Authorization": "Token token={{ .secrets.pagerduty_token }}",
+        "Accept": "application/vnd.pagerduty+json;version=2",
+        "Content-Type": "application/json",
+        "From": "{{ .trigger.by.user.email }}"
```
Comment (PeyGis) [inline integrations/pagerduty/.port/resources/actions.json:41]: is the `From` part of the acceptable PagerDuty header properties? I'm curious if this is to show who the sender of the incident is from the PagerDuty dashboard

File: integrations/pagerduty/.port/resources/actions.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
+[
+  {
+    "identifier": "trigger_incident",
+    "title": "Trigger incident",
```
Comment (PeyGis) [inline integrations/pagerduty/.port/resources/actions.json:4]:
```suggestion
    "title": "Trigger Incident",
```

File: integrations/pagerduty/.port/resources/actions.json [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            "title": "Description",
+            "description": "Description of the incident"
+          }
+        },
+        "required": [],
+        "order": [
+          "title"
+        ]
+      },
+      "blueprintIdentifier": "pagerdutyService"
```
Comment (PeyGis) [inline integrations/pagerduty/.port/resources/actions.json:28]: Why are we using service instead of incident?

**Review by PeyGis — CHANGES_REQUESTED** [review]
left a comment

File: integrations/pagerduty/.port/resources/actions.json (hunk 4 above)
Comment (oiadebayo) [reply integrations/pagerduty/.port/resources/actions.json:28]: A new incident needs to be created with regards to a service whose Id is on the service entity. We are not modifying an existing incident; the action needs to be performed on the service blueprint for us to create an incident about it

**Pull request closed without merging**